smg-automotive/components-pkg
# feat(Footer): Implements the footer as a component

References https://autoricardo.atlassian.net/browse/VSST-431
Design: https://www.figma.com/file/8vy6tgk0DV9KSKyJBjiLSH/12.-AS24---Web---Patterns?node-id=385%3A8908

## Motivation and context

Implements the footer as a chakra component.
The current links, translations and configurations have been extracted and added to the component configuration.
We might wanna migrate that to a CMS at some point in the future.

Interface:

- Brand: 'as24' | 'ms24': Controls weather to show the configuration variation for AutoScout24 or MotoScout24
- Language: de|fr|it|en
- useAbsoluteUrls: boolean: Controls whether to render the internal links as absolute URLs or relative paths. This will allow us to use the component on projects that have their own subdomain like `dealer.autoscout24.ch` 
- environment: Controls the Domain for absolute URLs so we don't end up in production from dev and preprod environments.

![image](https://user-images.githubusercontent.com/3371760/202700681-73449744-02e4-4498-99f7-6334c9521f00.png)

Note: We have a special case with the `Manage Preferences` which opens the OneTrust preference centre. The link has no effect when the component is used standalone

Regarding the links: I replaced all links being redirected with the resolved links and added a target blank to all external ones. Most the English links are linking to germans as the legacy apps do not support English.

## Before

N/A

## After

New component 🎉 

## How to test

https://lkappeler-vsst-431-footer.components-pkg.branch.autoscout24.ch/?path=/story/patterns-navigation-footer--footer

For **every language** and **every brand** ensure every link works and points towards the right target.
Ensure all the translations match the current production environments.
